pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -253,8 +259,18 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 	errList = append(errList, r.remediate(ctx, needRemediationTargets, mhc)...)
+
+	// remove external remediation annotation for healthy machines in case they get healthy before remediation happened
+	for _, h := range currentHealthy {
```

Comment:
I don't believe there's any safe way to treat a Machine that is potentially in the middle of being rebooted as 'healthy'. We need to treat it as unhealthy and apply the MaxUnhealthy limit as in #898, otherwise there will always be a possibility of remediating too many Machines at once.

It's possible to return a _separate_ list of "reprieved" Machines and try to cancel remediation as a backup. However, per https://github.com/openshift/cluster-api-provider-baremetal/pull/157#discussion_r683473906 I'm not sure there is ever a safe way to implement this.